.github/workflows/main.yaml

```diff
@@ -140,50 +143,139 @@ jobs:
       - name: Set up dependencies for Mac OS
         run: brew install automake
         if: matrix.os == 'macos-latest'
-      - run: make test
-      - name: Build and Test C API
-        run: |
-          make build-capi
-          make test-capi
-        if: matrix.os != 'windows-latest'
-      - name: Build C API on Windows
-        run: make build-capi
-        if: matrix.os == 'windows-latest'
+        # speed up iteration:
+    #  - run: make test
+    #  - name: Build and Test C API
+    #    run: |
+    #      make build-capi
+    #      make test-capi
+    #    if: matrix.os != 'windows-latest'
+    #  - name: Build C API on Windows
+    #    run: make build-capi
+    #    if: matrix.os == 'windows-latest'
       - name: Build Wasmer binary
         run: |
           make build-wasmer
       - name: Build Wapm binary
         run: |
           make build-wapm
         if: needs.setup.outputs.DOING_RELEASE == '1'
-      - name: Package Wasmer for integration tests
-        run: make package-without-wapm-for-integration-tests
-        if: needs.setup.outputs.DOING_RELEASE != '1'
-      - name: Package Wasmer
-        run: |
-          make package
-        if: needs.setup.outputs.DOING_RELEASE == '1'
-      - name: Run integration tests (Windows)
-        shell: cmd
+        # speed up iteration:
+      #- name: Package Wasmer for integration tests
+      #  run: make package-without-wapm-for-integration-tests
+      #  if: needs.setup.outputs.DOING_RELEASE != '1'
+      #- name: Package Wasmer
+      #  run: |
+      #    make package
+      #  if: needs.setup.outputs.DOING_RELEASE == '1'
+      #- name: Run integration tests (Windows)
+      #  shell: cmd
+      #  run: |
+      #    call refreshenv
+      #    set WASMER_DIR=%CD%\package
+      #    make test-integration
+      #  if: matrix.run_integration_tests && matrix.os == 'windows-latest'
+      #- name: Run integration tests (Unix)
+      #  run: |
+      #    export WASMER_DIR=`pwd`/package
+      #    make test-integration
+      #  if: matrix.run_integration_tests && matrix.os != 'windows-latest'
+      - name: Cross compile from Linux
+        if: matrix.os == 'ubuntu-18.04'
+        shell: bash
         run: |
-          call refreshenv
-          set WASMER_DIR=%CD%\package
-          make test-integration
-        if: matrix.run_integration_tests && matrix.os == 'windows-latest'
-      - name: Run integration tests (Unix)
+          ls target/release
+          alias wasmer=target/release/wasmer
+          ./target/release/wasmer compile --target=x86_64-apple-darwin tests/integration/cli/assets/qjs.wasm -o qjs_mac_from_linux.wjit
+          #./target/release/wasmer compile --target=x86_64-pc-windows-msvc tests/integration/cli/assets/qjs.wasm -o qjs_win_from_linux.wjit
+          mkdir cross
+          cp qjs_mac_from_linux.wjit cross
+          #cp qjs_win_from_linux.wjit cross
+          cp target/release/wasmer cross
```

Comment:
why not using the artifact uploaded in "Upload Artifacts" (we would have to remove this line: 
https://github.com/wasmerio/wasmer/pull/1779/files#diff-71cabc4177e41ea8f15a89eb65398fa8187739f68a2762706d84885cd8b2ddc3R225)?

Reply:
The cross compilation just needs a Wasmer binary and the cross-compiled binaries, the full release requires the C API and wapm.

I think it could be made to work but I think it's best to keep them separate to make it clear that they're not coupled.  If we change CI to be more parallel in the future, that might simplify things